### ISSUE TITLE
No error reported for specs that do not exist

### ISSUE THREAD

**Issue by pombredanne** [nexB/python-inspector#64]
See 
```
$ python-inspector --spec "foobar-does-not-exists" --json -
{
  "headers": {
    "tool_name": "python-inspector",
    "tool_homepageurl": "https://github.com/nexB/python-inspector",
    "tool_version": "0.6.5",
    "options": [
      "--specifier foobar-does-not-exists",
      "--index-url https://pypi.org/simple",
      "--python-version 38",
      "--operating-system linux",
      "--json <file>"
    ],
    "notice": "Dependency tree generated with python-inspector.\npython-inspector is a free software tool from nexB Inc. and others.\nVisit https://github.com/nexB/scancode-toolkit/ for support and download.",
    "warnings": [],
    "errors": []
  },
  "requirements": [
    {
      "purl": "pkg:pypi/foobar-does-not-exists",
      "extracted_requirement": "foobar-does-not-exists",
      "scope": "install",
      "is_runtime": true,
      "is_optional": false,
      "is_resolved": false,
      "resolved_package": {},
      "extra_data": {}
    }
  ],
  "resolved_dependencies": null
}
```

@aspacca ping

**Comment by aspacca** [comment]
@pombredanne is there a workaraound to indentify such case, like `resolved_package` being empty?

If so I could use https://github.com/nexB/pip-requirements-parser since now, instead of having to rely on pip and change the dependecy later on

**Comment by pombredanne** [comment]
@aspacca
>  was assuming that pip-requirements-parser depends on python-inspector, but indeed it does not seems so :)

that's the other way around:  python-inspector depends on pip-requirements-parser

**Comment by pombredanne** [comment]
@TG1999 ^

**Comment by aspacca** [comment]
@pombredanne 

I think where was a misunderstanding when I reported the issue on gitter :)

I mentioned about `pip-requirements-parser` not failing on non existing specs.


here's your feedback from back then:
```
pip requirements parser... it does parse the same way that pip parses. But it does NOT lookup in remote PyPI repos .
See https://github.com/nexB/python-inspector for this IMHO
parser-wise there is nothing to do there as we cannot know if a requirement exists or not.

The inspector does reuse pip-requirements-parser and resolve deps and interacts with PyPI repos alright and this would be the place IMHO.
I created [nexB/python-inspector#64](https://github.com/nexB/python-inspector/issues/64) ...
```

now I understand what you meant:
`pip-requirements-parser` does NOT lookup in remote PyPI repos, so the fix has to happen on `python-inspector`. 

the problem for me is that `python-inspector` is a cli command, not a package I can consume from my code.

unless there is a way to call `python-inspector` from code (preferably in an non-hacky way, like I have to do with `pip`, that's my current solution :)), I have to use `pip-requirements-parser`, and in this case the issue about having an error for specs that do not exists in a package still exists.


to further clarify my needs:
> The pip-requirements-parser library offers a simple and stable code API that will not change without notice.

I'd need a "simple and stable code API that will not change without notice" that reflects the resolution of version of the packages installed as done my the real pip.

this cannot be done with `pip-requirements-parser` because of the following:
> The pip-requirements-parser library works entirely offline and the requests dependency and calling has been entirely removed.

basically: am I supposed to invoke https://github.com/nexB/python-inspector/blob/main/src/python_inspector/api.py#L68 directly? :)

**Comment by pombredanne** [comment]
Here is an example https://github.com/nexB/scancode.io/blob/main/scanpipe/pipes/resolve.py#L42
At this stage we cannot promise a stable code API yet, but this tool is already used as CLI or Python library in a few places and therefore there will not be API breaking changes that are not properly managed and announced, with ahead of time deprecation warnings.
